.circleci/config.yml

```diff
@@ -343,100 +470,270 @@ jobs:
           command: |
             curl -Lo - https://github.com/goreleaser/goreleaser/releases/download/v1.4.1/goreleaser_Linux_x86_64.tar.gz | sudo tar -C /usr/local/bin -xvzf - goreleaser
       - run: etc/redhat/push_images.sh
+  pachyderm-prerelease-testing:
+    resource_class: large
+    machine:
+      image: << pipeline.parameters.machine_image >>
+    steps:
+      - checkout
+      - attach_workspace:
+          at: ./
+      - run:
+          name: "test examples"
+          command: |
+            sudo chmod +x etc/testing/circle/run_release_tests.sh
+            etc/testing/circle/run_release_tests.sh
+          no_output_timeout: 1h
+      - store_artifacts:
+          path: /tmp/debug-dump
+          destination: debug-dump
+  release-github:
+    docker:
+      - image: cimg/base:stable
+    steps:
+      - checkout
+      - attach_workspace:
+          at: ./
+      - gh/setup:
+          version: 2.13.0
+      - run:
+          name: create github release
+          command: gh release create ${CIRCLE_TAG} -t ${CIRCLE_TAG} ./dist-pach/pachctl/pachctl_${CIRCLE_TAG}* --generate-notes
+  release-docker-hub:
+    docker:
+      - image: cimg/base:stable
+    steps:
+      - checkout
+      - run:
+          name: pull docker images
+          command: |
+            make VERSION=$CIRCLE_SHA1 docker-pull
+      - run:
+          name: retag and push docker images
+          command: |
+            make VERSION=$VERSION docker-push
+  release-hombrew-tap:
+    docker:
+      - image: cimg/go:1.17.3
+    steps:
+      - checkout
+      - add_ssh_keys:
+          fingerprints:
+            - "4c:1c:ff:56:c1:d0:07:03:23:08:08:2e:e6:16:30:1d"
+      - run:
+          name: setup pachydermbuildbot git
+          command: |
+            git config --global user.email "buildbot@pachyderm.io"
+            git config --global user.name "pachydermbuildbot"
+      # TODO make this depend on tag regex
+      - run:
+          name: update homebrew formula
+          command: |
+            ./etc/build/update_homebrew.sh "alpha"
```

Comment:
no this needs a bit more thought. should be empty if its a regular patch release so that crazy homebrew script in the homebrew repo can index the correct tap. refactoring how we do homebrew releases will be one the first ones to address. its pretty simple to bend it to our will for the time being.

Reply:
So, we'll need to remove `alpha` before merging this then?

Reply:
so the existing update homebrew is hacky, but I don't want to touch it right now. aslong as this gets a null/blank value for only patch releases it does the expected result. so we'll just use the additional value on the tag which will have a value for every time we do a (custom, alpha, prerelease), and have a blank value on patch releases. and since we only trigger this on the regex tag alais should be protection enough for now.